src/main/java/io/cdap/plugin/http/source/batch/HttpBatchSource.java

```diff
@@ -88,11 +92,13 @@ public void prepareRun(BatchSourceContext context) {
       .setFqn(config.getUrl()).build();
     LineageRecorder lineageRecorder = new LineageRecorder(context, asset);
     lineageRecorder.createExternalDataset(schema);
-    lineageRecorder.recordRead("Read", String.format("Read from HTTP '%s'", config.getUrl()),
-      Preconditions.checkNotNull(schema.getFields()).stream()
-        .map(Schema.Field::getName)
-        .collect(Collectors.toList()));
+    List<String> getNameList = schema != null ? Preconditions.checkNotNull(schema.getFields()).stream()
+            .map(Schema.Field::getName)
+            .collect(Collectors.toList()) : new ArrayList<>();
```

Comment:
wrap this in `try-catch`, if there is an `NPE`, we should throw proper `ProgramFailureException`.